cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedStreamSizeScheduleStore.java

```diff
@@ -232,9 +235,7 @@ public void apply() throws Exception {
       .execute(new TransactionExecutor.Subroutine() {
         @Override
         public void apply() throws Exception {
-          byte[] startKey = Bytes.toBytes(KEY_PREFIX);
-          byte[] endKey = Bytes.stopKeyForPrefix(startKey);
-          Scanner scan = table.scan(startKey, endKey);
+          Scanner scan = getScannerWithPrefix(KEY_PREFIX);
```

Comment:
Use a `try (Scanner scanner = getScannerWithPrefix(...)) { }` block to make sure that.